### PR TITLE
fix bug in case the A or AAAA records is processed before SRV record

### PR DESCRIPTION
**Pull request by stefaneicher**
Hi Kay,

I had an error using the library. see ServiceInfoImplTest.java
the error occurs if  you uncomment this row
allAnswers = aRecordsLast(allAnswers);

Gruss Stefan

### CODE REVIEW

File: src/test/java/javax/jmdns/impl/ServiceInfoImplTest.java [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+                "DeviceManagementService",
+                "",
+                -1,
+                -1,
+                -1,
+                true,
+                new byte[]{});
+        jmDNS.addListener(serviceInfo, null);
+        jmDNS.handleResponse(msg);
+
```
Comment (kaikreuzer) [inline src/test/java/javax/jmdns/impl/ServiceInfoImplTest.java]: You should close jmDNS here again.

File: src/test/java/javax/jmdns/impl/ServiceInfoImplTest.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.net.DatagramPacket;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+
+/**
+ * @author stefan eicher https://github.com/stefaneicher
```
Comment (kaikreuzer) [inline src/test/java/javax/jmdns/impl/ServiceInfoImplTest.java]: Please just do `@author Stefan Eicher`

File: src/test/java/javax/jmdns/impl/ServiceInfoImplTest.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+        assertEquals(serviceInfo.getWeight(), 0);
+        assertEquals(serviceInfo.getPriority(), 0);
+        assertEquals(serviceInfo.isPersistent(), true);
+
+        // ... the ip address is set
+        assertEquals(serviceInfo.getInet4Addresses().length, 1);
+        assertArrayEquals(serviceInfo.getInet4Addresses()[0].getAddress(), new  byte[]{(byte) 192, (byte) 168,88, (byte) 236});
+    }
+
+    @SuppressWarnings("ResultOfMethodCallIgnored")
```
Comment (kaikreuzer) [inline src/test/java/javax/jmdns/impl/ServiceInfoImplTest.java]: what is this good for? This gives a warning in my IDE...

**Comment by kaikreuzer** [comment]
Thanks Stefan,

Unfortunately, your test fails for me:
```
Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.011 sec <<< FAILURE! - in javax.jmdns.impl.ServiceInfoImplTest
test_ip_address_is_set(javax.jmdns.impl.ServiceInfoImplTest)  Time elapsed: 0.01 sec  <<< ERROR!
java.io.IOException: DNSIncoming corrupted message
	at javax.jmdns.impl.DNSIncoming.<init>(DNSIncoming.java:199)
	at javax.jmdns.impl.ServiceInfoImplTest.test_ip_address_is_set(ServiceInfoImplTest.java:21)
```
Any idea what could be wrong?

Please also make sure to [sign off your commit](https://github.com/jmdns/jmdns/blob/master/CONTRIBUTING.md#sign-your-work), which is required for contributions.

File: src/test/java/javax/jmdns/impl/ServiceInfoImplTest.java (hunk 3 above)
Comment (stefaneicher) [inline src/test/java/javax/jmdns/impl/ServiceInfoImplTest.java]: I use Intellij ultimate and I want to handle all warnings the IDE gives me to let other peoble know I addressed them. Unfortentualy eclipse does not understand them.

image

image

**Comment by stefaneicher** [comment]
Sorry something went woring as I check in the binary file. It is updated now.
Also see the wiresahrk file a_record_before_srv.pcapng for more details.

image

File: src/test/java/javax/jmdns/impl/ServiceInfoImplTest.java (hunk 3 above)
Comment (kaikreuzer) [inline src/test/java/javax/jmdns/impl/ServiceInfoImplTest.java]: I see. How great - to make a warning disappear in one IDE adds one in another...
Well, fine for me to leave it as is.

File: src/main/java/javax/jmdns/impl/JmDNSImpl.java [hunk 4]
```diff
@@ -1429,6 +1412,48 @@ void handleResponse(DNSIncoming msg) throws IOException {
...
+     * PassengerCountingService._ibisip_http._tcp.local: type TXT, class IN, cache flush
+     * DIST500_7-F07_OC030_05_03941.local: type A, class IN, cache flush, addr 192.168.88.236
+     * DeviceManagementService._ibisip_http._tcp.local: type SRV, class IN, cache flush, priority 0, weight 0, port 5000, target DIST500_7-F07_OC030_05_03941.local
+     * PassengerCountingService._ibisip_http._tcp.local: type SRV, class IN, cache flush, priority 0, weight 0, port 5001, target DIST500_7-F07_OC030_05_03941.local
+     * DeviceManagementService._ibisip_http._tcp.local: type NSEC, class IN, cache flush, next domain name DeviceManagementService._ibisip_http._tcp.local
+     * PassengerCountingService._ibisip_http._tcp.local: type NSEC, class IN, cache flush, next domain name PassengerCountingService._ibisip_http._tcp.local
+     * DIST500_7-F07_OC030_05_03941.local: type NSEC, class IN, cache flush, next domain name DIST500_7-F07_OC030_05_03941.local
+     */
+    private List<DNSRecord> aRecordsLast(List<DNSRecord> allAnswers) {
+        ArrayList<DNSRecord> ret = new ArrayList<DNSRecord>();
```
Comment (kaikreuzer) [inline src/main/java/javax/jmdns/impl/JmDNSImpl.java]: To slightly optimize it, I would suggest to change it to
```
        List<DNSRecord> ret = new ArrayList<DNSRecord>(allAnswers.size());
```

**Comment by kaikreuzer** [comment]
Hi Stefan,

Thanks for the updates, looks good to me now!
Before I can merge, may I ask you to squash your commits into one and [sign-off the resulting commit](https://github.com/jmdns/jmdns/blob/master/CONTRIBUTING.md#sign-your-work)? Many thanks!

**Comment by stefaneicher** [comment]
Hallo Kai, sicher doch. I will do it on monday. In case you just merge my
commit I will try to make it even nicer (undo my intellij formatting...).
Gruss Stefan

Kai Kreuzer <notifications@github.com> schrieb am Fr., 29. Juli 2016 16:53:

**Comment by kaikreuzer** [comment]
I will wait for your "go" and merge then as-is. Schönes Wochenende!

**Comment by stefaneicher** [comment]
Hi kai, I squashed all to one commit. I would highly appreciate if you
could release a new version with this bug fix. Can you do that?

Kai Kreuzer <notifications@github.com> schrieb am Fr., 29. Juli 2016 17:02:

File: src/main/java/javax/jmdns/impl/JmDNSImpl.java (hunk 4 above)
Comment (kaikreuzer) [inline src/main/java/javax/jmdns/impl/JmDNSImpl.java]: Did you see this comment?

**Comment by kaikreuzer** [comment]
Hi Stefan,

One comment left (https://github.com/jmdns/jmdns/pull/80#discussion_r72804651) to answer and your commit is still not signed-off.
I will certainly do a new release with this fix, the question is when ;-)

**Comment by stefaneicher** [comment]
Hi Kai,
I added the sing-of, removed the intellij files and added the
"List<DNSRecord> ret = new ArrayList<DNSRecord>(allAnswers.size());"

2016-08-02 10:59 GMT+02:00 Kai Kreuzer <notifications@github.com>:

**Comment by kaikreuzer** [comment]
Many thanks!

Now I have to ask you for a favor: Could you briefly review my PR https://github.com/jmdns/jmdns/pull/86 and comment if it is ok to merge?

**Comment by stefaneicher** [comment]
Off course. I will today or if I can tomorrow

Kai Kreuzer <notifications@github.com> schrieb am Di., 2. Aug. 2016 17:42:

**Pull request merged**